source/simulator/nullspace.cc

```diff
@@ -108,6 +220,56 @@ namespace aspect
   template <int dim>
   void Simulator<dim>::setup_nullspace_constraints(AffineConstraints<double> &constraints)
   {
+
+    if (parameters.nullspace_removal & Parameters<dim>::NullspaceRemoval::any_rotation)
+      {
+        const bool is_spherical = Plugins::plugin_type_matches<const GeometryModel::SphericalShell<dim>>(*geometry_model)
+                                  || Plugins::plugin_type_matches<const GeometryModel::Sphere<dim>>(*geometry_model);
+
+        Assert(is_spherical,
+               ExcNotImplemented("nullspace constraints for rotation are currently only implemented Sphere and "
+                                 "SphericalShell geometries."));
```

Comment:
This will probably trigger warnings in release model that `is_spherical` is not used.